sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -87,6 +87,7 @@ class Analyzer(
       EliminateUnions),
     Batch("Resolution", fixedPoint,
       ResolveRelations ::
+      SubstituteHints ::
```

Comment:
That's a good point. What is the best policy for a `Rule` with a fixedPoint `Once`?
To minimize the side effect of `Hint` for other rules, I put it at the very next of `ResolveRelation`.